mods/health/hud.lua

```diff
@@ -182,76 +233,116 @@ local function color_envirotemp(v, meta)
 	local danger_high = stress_high +40
 	local overlay
 
-	local col = 0xFFFFFF
+	local stat_col = stat_fine
+	local ttype = "hud_temp_normal"
 
 	if v > danger_high or v < danger_low then
-		col = 0x8008FF
+		icon_col = "extreme"
+		stat_col = stat_extreme
+		if v > danger_high then ttype = "hud_temp_hot" end
+		if v < danger_low then ttype = "hud_temp_cold" end
 	elseif v > stress_high or v < stress_low then
-		col = 0xDF0000
+		stat_col = stat_major
+		if v > stress_high then ttype = "hud_temp_hot" end
+		if v < stress_low then ttype = "hud_temp_cold" end
 	elseif v > comfort_high or v < comfort_low then
-		col = 0xFDFF46
+		stat_col = stat_slight
+		if v > comfort_high then ttype = "hud_temp_hot" end
+		if v < comfort_low then ttype = "hud_temp_cold" end
 	end
 	if v < stress_low then
 	   overlay = "weather_hud_frost.png"
+	   ttype = "hud_temp_cold"
```

Comment:
Setting the ttype alongside the overlay is superfluous, we can probably drop lines 255 and 259.

Reply:
Removed them.